tags.go

```diff
@@ -0,0 +1,47 @@
+package distribution
+
+// Tag is a named reference to a repository object. Target generally refers to
+// a manifest.
+type Tag interface {
+	Blob // io access to serialized content.
+
+	// Name returns the fully qualified name of the object. For example, if
+	// the object is in repository "stevvooe/test" and has tag "latest", the
+	// name will be "stevvooe/test:latest".
+	Name() string
+
+	// Description provides an overview of the semantic meaning of this
+	// particular tag. This could include release notes or documentation that
+	// are helpful to the user.
+	Description() string
+
+	// Meta exposes free-form data associated with this tag.
+	Meta() map[string]interface{}
+
+	// Target identifies the blob targeted by this tag.
+	Target() Descriptor
+}
+
+// TagService provides access to information about tagged objects.
+type TagService interface {
```

Comment:
Let's simplify the tag service to be a mapping between tag name to a `Descriptor`.